spring-data-jpa/src/main/java/org/springframework/data/jpa/repository/support/DefaultJpaContext.java

```diff
@@ -61,17 +61,12 @@ public EntityManager getEntityManagerByManagedType(Class<?> type) {
 
 		Assert.notNull(type, "Type must not be null");
 
-		if (!entityManagers.containsKey(type)) {
-			throw new IllegalArgumentException(String.format("%s is not a managed type", type));
-		}
+		Assert.isTrue(entityManagers.containsKey(type), () -> String.format("%s is not a managed type", type));
 
 		List<EntityManager> candidates = this.entityManagers.get(type);
 
-		if (candidates.size() == 1) {
-			return candidates.get(0);
-		}
-
-		throw new IllegalArgumentException(
-				String.format("%s managed by more than one EntityManagers: %s", type.getName(), candidates));
+		Assert.isTrue(candidates.size() == 1, () -> String.format("%s managed by more than one EntityManagers: %s", type.getName(), candidates));
```

Comment:
One thought for here: The control flow is being replaced with calls to `Assert`. While assertions are fine for method-entry checks, usage here is kind-of unexpected. Also, we're always creating a capturing lambda leading to a bit more GC pressure.